### PR TITLE
Move Sophtron adapter into own repo

### PR DESCRIPTION
**Pull request by tphalp**
Sophtron Adapter

### CODE REVIEW

File: apps/server/src/adapterSetup.ts [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
-};
+const sophtronAdapterMapObject: Record<string, AdapterMap> =
+  getSophtronAdapterMapObject({
+    logClient: logger,
+    aggregatorCredentials: {
+      clientId: config.SophtronApiUserId,
+      secret: config.SophtronApiUserSecret,
+      endpoint: config.SophtronApiServiceEndpoint,
+      vcEndpoint: config.SophtronVCServiceEndpoint,
+      aggregator: "sophtron",
```
Comment (wesrisenmay-mx) [inline apps/server/src/adapterSetup.ts]: Why is "aggregator" a variable?

File: apps/server/src/adapterSetup.ts [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+      available: true,
```
Comment (wesrisenmay-mx) [inline apps/server/src/adapterSetup.ts]: What is available for?

File: apps/server/src/adapterSetup.ts [hunk 3]
```diff
@@ -46,12 +56,12 @@ const mxAdapterMapObject = getMxAdapterMapObject({
...
-export const adapterMap = {
+export const adapterMap: Record<string, AdapterMap> = {
   ...templateAdapterMapObject,
   ...mxAdapterMapObject,
   ...sophtronAdapterMapObject,
   ...testAdapterMapObject,
 };
 
-export type Aggregator = keyof typeof adapterMap;
+export type Aggregator = keyof typeof adapterMap | string;
```
Comment (wesrisenmay-mx) [inline apps/server/src/adapterSetup.ts]: Why the | string?

File: packages/sophtron-adapter/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+```typescript
+const sophtronAdapterMapObject = getSophtronAdapterMapObject({
+    logClient: logger,
+    aggregatorCredentials: {
+        sophtron: {
+            username: config.MxClientId,
+            password: config.MxApiSecret,
+            basePath: 'https://int-api.mx.com',
+            vcEndpoint: 'https://int-api.mx.com/',
+            aggregator: 'sophtron',
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/README.md]: Don't forgot to update these if we remove them

File: packages/sophtron-adapter/src/createVc.test.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { VCDataTypes } from "@repo/utils";
+
+import type { AdapterDependencies } from "./models";
+import { logClient } from "./test/utils/logClient";
+import { aggregatorCredentials } from "./adapter.test";
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/createVc.test.ts]: IMO we shouldn't be importing things from test files

File: packages/sophtron-adapter/src/getVc.test.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { logClient } from "./test/utils/logClient";
+import { sophtronVcTranscationsData } from "./test/testData/sophtronVcData";
+import { aggregatorCredentials } from "./adapter.test";
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/getVc.test.ts]: Same thing here

File: packages/sophtron-adapter/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import { createSophtronVC } from "./createVc";
+import type { AdapterDependencies } from "./models";
+import { SOPHTRON_ADAPTER_NAME } from "./constants";
+
+export const getSophtronAdapterMapObject = (
+  dependencies: AdapterDependencies,
+) => {
+  return {
+    sophtron: {
+      testInstitutionAdapterName: SOPHTRON_ADAPTER_NAME,
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/index.ts]: This variable is if you need to route to something besides the name of the provider. No need to have it here.

File: packages/sophtron-adapter/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      vcAdapter: createSophtronVC(dependencies),
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/index.ts:14]: We need a data adapter here

File: packages/sophtron-adapter/src/test/testData/auth0.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export const accessTokenResponse = {
+  access_token:
+    'eyJhbGciOiJSUzI1NiIsInR5323324wwTlV3MktRaWZfZVNrR3Y3M1FrMyJ9.eyJpc3MiOiJodHRwczovL2Rldi1kMjN3YXU4bzB1YzVodzhuLnVzLmF1dGgwLmNvbS8iLCJzdWIiOiJVMVM1cjVFUTliUHFYRDVhaTY3Njl1OW9pNjNCRDFTMkBjbGllbnRzIiwiYXVkIjoidWNwLXdpZGdldC1pbnRlcmFjdGlvbnMiLCJpYXQiOjE3MjUzOTM1MzgsImV4cCI6MTcyNTQ3OTkzOCwic2NvcGUiOiJyZWFkOndpZGdldC1lbmRwb2ludHMgd3JpdGU6d2lkZ2V0LWVuZHBvaW50cyIsImd0eSI6ImNsaWVudC1jcmVkZW50aWFscyIsImF6cCI6IlUxUzVyNUVROWJQcVhENWFpNjc2OXU5b2k2M0JEMVMyIiwicGVybWlzc2lvbnMiOlsicmVhZDp3aWRnZXQtZW5kcG9pbnRzIiwid3JpdGU6d2lkZ2V0LWVuZHBvaW50cyJdfQ.SM7Z37vxfRVDELg2ohi3_ei4973hHGJ5NR44-i7b-7Vqh3RB_p0J-E7H5k1D3T2qH-5Wk93xHQKvd1CN1uDflvwhCHh7QMDLUJu1s1E8J3ifTuy72Wbz5-PpSA52dWDuvjtHVxUO5fXQYrgi3sOEVksromgrga9NzD2_C0ZFQLnrg3TrfRP5AC3stInCdWqxThA5Res116kKzbXohpLOX5poIz_b5vYB_rCd6HcfDb1BWgNwrbNlE87b9z6yzyKj_TTg40UjnwavxjryZW7RtNEOulCiSxp1iv37Hp4f4MjcnUDPOJesJWK4bAdfE687VfshbnVXedelRamAY254bA',
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/test/testData/auth0.ts]: Is this safe to commit?

File: apps/server/src/adapterSetup.ts (hunk 1 above)
Comment (tphalp) [reply apps/server/src/adapterSetup.ts]: I'm sure I just copied these from somewhere else. I didn't add that explicitly.

File: apps/server/src/adapterSetup.ts (hunk 2 above)
Comment (tphalp) [reply apps/server/src/adapterSetup.ts]: I'm sure I just copied these from somewhere else. I didn't add that explicitly.

File: packages/sophtron-adapter/src/test/testData/auth0.ts (hunk 9 above)
Comment (tphalp) [reply packages/sophtron-adapter/src/test/testData/auth0.ts]: I've removed this. It was originally in apps/server.

File: packages/sophtron-adapter/README.md (hunk 4 above)
Comment (tphalp) [reply packages/sophtron-adapter/README.md]: Removed

File: packages/sophtron-adapter/src/index.ts (hunk 8 above)
Comment (tphalp) [reply packages/sophtron-adapter/src/index.ts:14]: Added

File: apps/server/src/adapterSetup.ts (hunk 3 above)
Comment (tphalp) [reply apps/server/src/adapterSetup.ts]: Without this I was getting type errors while running some tests. I've just removed it, and it seems things are back to normal.

File: packages/sophtron-adapter/src/test/testData/auth0.ts (hunk 9 above)
Comment (tphalp) [reply packages/sophtron-adapter/src/test/testData/auth0.ts]: It is an invalid token, as it's not properly signed by Auth0, so I don't think it matters that it's already in the repo.

File: packages/sophtron-adapter/src/test/testData/credentials.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { SOPHTRON_ADAPTER_NAME } from "../../constants";
+import type { ApiCredentials } from "../../models";
+
+export const aggregatorCredentials: ApiCredentials = {
+  clientId: "testClientId",
+  secret: "testSecret",
+  endpoint: "https://api.sophtron.com/api",
+  vcEndpoint: "https://vc.sophtron.com/api/",
+  aggregator: SOPHTRON_ADAPTER_NAME,
```
Comment (wesrisenmay-mx) [inline packages/sophtron-adapter/src/test/testData/credentials.ts]: 💣

**Review by wesrisenmay-mx — DISMISSED** [review]

**Review by wesrisenmay-mx — APPROVED** [review]

**Pull request merged**